chia/full_node/full_node_api.py

```diff
@@ -46,17 +47,20 @@
 from chia.types.unfinished_block import UnfinishedBlock
 from chia.util.api_decorators import api_request, peer_required, bytes_required, execute_task, reply_type
 from chia.util.full_block_utils import header_block_from_block
-from chia.util.generator_tools import get_block_header
+from chia.util.generator_tools import get_block_header, tx_removals_and_additions
 from chia.util.hash import std_hash
 from chia.util.ints import uint8, uint32, uint64, uint128
 from chia.util.merkle_set import MerkleSet
+from chia.full_node.mempool_check_conditions import get_name_puzzle_conditions
 
 
 class FullNodeAPI:
     full_node: FullNode
+    executor: ThreadPoolExecutor
 
     def __init__(self, full_node: FullNode) -> None:
         self.full_node = full_node
```

Comment:
It might be better to put this in Full Node. Full Node Api is just a wrapper class and shouldn't have any state

Reply:
yeah, I was considering that. It's not clear cut in my mind. The API boundary of the full node isn't high level enough (today) to encompass this, I would say. There's a fair amount of work, and assembling, happening in the full_node_api.

Taken all the way, the `FullNodeAPI` perhaps should just be a protocol that `FullNode` implements (i.e. there's literally nothing left in `FullNodeAPI`). I get the impression the main purpose of the `API` classes are to provide limited views onto their underlying objects.

Reply:
They used to be both together, but I split them up because it was getting confusing to see where the API methods were. It's true that the boundaries are not clear cut though..